kernel/build-kernel.sh

```diff
@@ -136,19 +138,86 @@ get_major_kernel_version() {
 	echo "${major_version}.${minor_version}"
 }
 
+# Make a kernel config file from generic and arch specific
+# fragments
+# - arg1 - path to arch specific fragments
+# - arg2 - path to kernel sources
+#
+get_kernel_frag_path() {
+	local arch_path="$1"
+	local common_path="${arch_path}/../common"
+	local kernel_path="$2"
+	local cmdpath="${kernel_path}/scripts/kconfig/merge_config.sh"
+	local config_path="${arch_path}/.config"
+
+	local arch_configs="$(ls ${arch_path}/*.conf)"
+	local common_configs="$(ls ${common_path}/*.conf)"
+
+	# Later, if we need to add kernel version specific subdirs in order to
+	# handle specific cases, then add the path definition and search/list/cat
+	# here.
+	local all_configs="${common_configs} ${arch_configs}"
+
+	info "Constructing config from fragments: ${config_path}"
+	local results=$(export KCONFIG_CONFIG=${config_path}; export ARCH=${arch_target}; cd ${kernel_path}; ${cmdpath} -r -n ${all_configs})
+
+	# Did we request any entries that did not make it?
+	local missing=$(echo $results | grep -v -q "not in final"; echo $?)
```

Comment:
This `not in path` looks a bit "magic" to me. I'd be tempted to put that string into it's own variable with a juicy description to make it easier to fix it the day that tool decides to change its output format.

Reply:
it, and its two friends (the words 'redefined' and 'redundant') are only used once each - I'm not convinced pulling them into separate strings actually makes it any easier to read ('it is looking for what string?') or easier to update ('the error line does not have the string in it... where the *** is it...?'), but, as you wish...